### DISCUSSION TITLE
tell me how to fix

### DISCUSSION THREAD

**Discussion by corsairs** [azerothcore/azerothcore-wotlk#7720] [Tools]
How can I fix this myself?
https://github.com/azerothcore/azerothcore-wotlk/issues/7672

**Comment by lineagedr** [comment]
https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Entities/Unit/Unit.cpp#L2935
```diff
+static constexpr int8 SPELL_HIT_CAP_PERCENT = 5; // 5% hit chance
+bool canMiss = int8(m_modSpellHitChance) <= SPELL_HIT_CAP_PERCENT;
-if (rand < tmp)
+if ((rand < tmp) && !canMiss)
+    return SPELL_MISS_MISS;
```

**Reply by corsairs** [reply]
@lineagedr Why should you bet exactly 5%?

**Reply by Helias** [reply]
@lineagedr why don't open a PR?

**Reply by lineagedr** [reply]
@Helias Because it's a hackfix and it's not blizzlike

**Reply by Helias** [reply]
ok, but if it's better than the current implementation I think that it is not an issue to put the hackfix

**Comment by corsairs** [comment]


**Reply by lineagedr** [reply]
of course it does.

**Reply by corsairs** [reply]
It doesn't work for me, maybe I need to add something else.

**Reply by lineagedr** [reply]
@corsairs it's only for spells not for melee attacks

above 5% hit rating

https://user-images.githubusercontent.com/12139798/132124648-55235230-a207-40fc-8648-0cb5b920359f.mp4

Below 5% hit rating

https://user-images.githubusercontent.com/12139798/132124654-7cd7faf0-c049-4c7f-a61e-4d9dd8fc2247.mp4

**Comment by corsairs** [comment]
thanks helped, figured out how to fix
